sycl/unittests/xpti_trace/xptitest_subscriber/XPTISubscriber.cpp

```diff
@@ -0,0 +1,66 @@
+//==---------------------- XPTISubscriber.cpp --------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "xpti/xpti_trace_framework.h"
+
+#include <deque>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <string_view>
+#include <thread>
+
+std::deque<std::pair<uint16_t, std::string>> GReceivedNotifications;
+XPTI_CALLBACK_API void testCallback(uint16_t TraceType,
+                                    xpti::trace_event_data_t * /*Parent*/,
+                                    xpti::trace_event_data_t *Event,
+                                    uint64_t /*Instance*/,
+                                    const void *UserData) {
+  if (TraceType == xpti::trace_diagnostics) {
```

Comment:
```suggestion
  if (TraceType != xpti::trace_diagnostics)
    return;

```

Early exist will make it more readable.

Reply:
done